src/appBootstrapper.jsx

```diff
@@ -21,6 +23,27 @@ const stores = {
     // ...other stores
 };
 
+const end = superagent.Request.prototype.end;
+
+let redirecting = false;
+
+superagent.Request.prototype.end = function (callback) {
+    return end.call(this, (error, response) => {
+        if (redirecting) {
+            return;
+        }
+        if (response.statusCode === 401) {
+            var storageKey = `redirect${Math.floor(Math.random() * 1000000000000)}`
+            localStorage.setItem(storageKey, window.location.hash);
+            const loginUrl = `${getHost()}/?spring-security-redirect=${window.location.pathname}${window.location.search}${encodeURIComponent('#/restore?key=' + storageKey)}`;
+            redirecting = true;
```

Comment:
set this flag so that we don't redirect twice due to concurrent calls which both 401.